src/sentry/coreapi.py

```diff
@@ -369,12 +380,17 @@ def parse_client_as_sdk(self, value):
         }
 
     def should_filter(self, project, data, ip_address=None):
-        # TODO(dcramer): read filters from options such as:
-        # - ignore errors from spiders/bots
-        # - ignore errors from legacy browsers
         if ip_address and not is_valid_ip(ip_address, project):
             return True
 
+        release = data.get('release')
+        if release and not is_valid_release(release, project):
+            return True
+
+        error_message = data.get('sentry.interfaces.Message', {}).get('message')
+        if error_message and not is_valid_error_message(error_message, project):
```

Comment:
Would this not have already gone through a `validate_data` method such that `error_message` could only be a string once it hits `should_filter`? I might not be wading through the function calls clearly enough to discern the entire flow or maybe I don't understand the risk. Would it be better for me to do the check here as well? Like:

```
if not isinstance(error_message, six.string_types):
            error_message = json.dumps(error_message)
```